test/test_cuda.py

```diff
@@ -1902,29 +1902,29 @@ def worker(rank):
 t2.start()
 """])
 
-
-class TestHandlesThreadSafety(TestCase):
-    # These tests could not be put into TestCuda because TestCuda has
-    # _do_cuda_non_default_stream = True
-    # which makes them fails
-
     @skipIfRocm
     def test_cublas_multiple_threads_same_device(self):
-        size = 16
-        weight = torch.ones((size, size), device='cuda')
+        # Note, these parameters should be very carefully tuned
+        # Too small number makes it hard for the racing condition
+        # to happen, while too large number sometimes cause hang
```

Comment:
How and why do hangs happen?

Reply:
With this PR, it always succeeds. 

Without this PR, I could observe hangs when you have `test_iters = 1000`. It does not consistently hang, and whenever it doesn't hang, it will fail. When it hangs, you could see it runs at 100% GPU usage for a long time. If we make `test_iters = 100` like it is right now, it could still consistently fail, but I never observe hang.

I don't know why the hang, seems to be a deadlock inside cublas? The hangs are also reported by user https://github.com/pytorch/pytorch/issues/6962#issuecomment-547198718, so I am not surprised about it.

Reply:
Ah, ok, if hangs don't happen with this PR then it's fine, they probably come from cublas because handles are not thread safe.